cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -0,0 +1,374 @@
+            config.put(configPrefix + SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, "${file:" + CONNECTORS_CONFIG_FILE + ":" + SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG + "}");
+        }
+
+        if (cluster.getAuthentication() != null) {
+            Map<String, String> authProperties = AuthenticationUtils.getClientAuthenticationProperties(cluster.getAuthentication());
+            if (authProperties.containsKey(AuthenticationUtils.SASL_MECHANISM)) {
+                if (cluster.getTls() != null) {
+                    securityProtocol = "SASL_SSL";
+                } else {
+                    securityProtocol = "SASL_PLAINTEXT";
+                }
+            }
+
+            if (cluster.getAuthentication() instanceof KafkaClientAuthenticationTls) {
+                config.put(configPrefix + SslConfigs.SSL_KEYSTORE_TYPE_CONFIG, "PKCS12");
+                config.put(configPrefix + SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, STORE_LOCATION_ROOT + cluster.getAlias() + KEYSTORE_SUFFIX);
+                config.put(configPrefix + SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, "${file:" + CONNECTORS_CONFIG_FILE + ":" + SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG + "}");
+            }
+
+            String jaasConfig = null;
+            String saslMechanism = null;
+            String clientAuthType = authProperties.get(AuthenticationUtils.SASL_MECHANISM);
+            if (KafkaClientAuthenticationPlain.TYPE_PLAIN.equals(clientAuthType)) {
+                saslMechanism = "PLAIN";
+                jaasConfig = "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"" + authProperties.get(AuthenticationUtils.SASL_USERNAME) + "\" password=\"${file:" + CONNECTORS_CONFIG_FILE + ":" + cluster.getAlias() + ".sasl.password}\";";                    
+            } else if (KafkaClientAuthenticationScramSha512.TYPE_SCRAM_SHA_512.equals(clientAuthType)) {
+                saslMechanism = "SCRAM-SHA-512";
+                jaasConfig = "org.apache.kafka.common.security.scram.ScramLoginModule required username=\"" + authProperties.get(AuthenticationUtils.SASL_USERNAME) + "\" password=\"${file:" + CONNECTORS_CONFIG_FILE + ":" + cluster.getAlias() + ".sasl.password}\";";
+            } else if (KafkaClientAuthenticationOAuth.TYPE_OAUTH.equals(clientAuthType)) {
+                KafkaClientAuthenticationOAuth oauth  = (KafkaClientAuthenticationOAuth) cluster.getAuthentication();
+
+                StringBuilder oauthJaasConfig = new StringBuilder();
+                oauthJaasConfig.append("org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required ");
+
+                if (authProperties.containsKey("OAUTH_CONFIG")) {
+                    oauthJaasConfig.append(authProperties.get("OAUTH_CONFIG"));
+                }
+
+                if (oauth.getClientSecret() != null) {
+                    oauthJaasConfig.append(" oauth.client.secret=\"${file:" + CONNECTORS_CONFIG_FILE + ":" + cluster.getAlias() + ".oauth.client.secret}\"");
```

Comment:
What happens if `cluster.getAlias()` returns a string containing a `"`, or some other character which would require quoting here? I guess we expect the alias to be an identifier. If so we should use a `@Pattern` on the property in the model to enforce that.

Reply:
Good idea, done. The MM2 cluster alias ends up as a prefix on the mirrored topic names, so it needs to be a valid Kafka name.